lib/controllers/facebook-login.js

```diff
@@ -60,21 +60,27 @@ module.exports = function (req, res) {
 
       res.locals.user = expandedAccount;
       req.user = expandedAccount;
-      helpers.createIdSiteSession(req.user, req, res);
 
-      var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
+      helpers.createStormpathSession(req.user, req, res, function (err) {
```

Comment:
The callbacks for all the social providers looks very similar. Actually the whole controllers looks very similar to each other. Any reason why they are not just one generic controller?

*I know this is not your fault, I'm just curious :)*